### PR TITLE
Validate there are no critical call edges in optimized MIR

### PR DESCRIPTION
**Pull request by tmiasko**

### CODE REVIEW

**Comment by bjorn3** [comment]
There is a typo in the last commit message: corotuine -> coroutine

File: compiler/rustc_const_eval/src/transform/validate.rs [hunk 1]
```diff
@@ -285,6 +285,12 @@ impl<'a, 'tcx> CfgChecker<'a, 'tcx> {
             UnwindAction::Unreachable | UnwindAction::Terminate(UnwindTerminateReason::Abi) => (),
         }
     }
+
+    fn is_critical_edge(&self, target: Option<BasicBlock>, unwind: UnwindAction) -> bool {
```
Comment (cjgillot) [inline compiler/rustc_const_eval/src/transform/validate.rs:289]: Could you rename or inline this method? It is specific to *call* edges, and does not detect critical edges due to a switchint.

**Review by cjgillot — COMMENTED** [review]
R=me with a nit.

**Pull request merged**